File: generate_image_map.js
const fs = require('fs');
const files = fs.readFileSync('generated_file_names.txt', 'utf8').split('\n').filter(Boolean);

const map = {};
files.forEach(f => {
    // extract base, up/down
    // e.g. card_ganjin_1771334875863_down_1772087377242.png
    let parts = f.split('_');
    let type = parts[parts.length - 2]; // up or down
    let base = parts.slice(0, parts.length - 2).join('_');
    
    if(!map[base]) map[base] = {};
    map[base][type] = f;
});

console.log(JSON.stringify(map, null, 2));

File: _legacy_archive/generate_image_map.py
import json
import os

files = [f for f in os.listdir("assets/images") if "up_" in f or "down_" in f and f.endswith(".png")]

mapping = {}
for file in files:
    parts = file.split('_')
    if 'up' in parts:
        idx = parts.index('up')
        type_str = 'up'
    elif 'down' in parts:
        idx = parts.index('down')
        type_str = 'down'
    else:
        continue
        
    base = "./assets/images/" + "_".join(parts[:idx]) + ".png"
    if base not in mapping:
        mapping[base] = {}
    mapping[base][type_str] = "./assets/images/" + file

with open("image_map.json", "w") as f:
    json.dump(mapping, f, indent=2)
    
print("Updated image_map.json")
